src/python/pants/backend/python/goals/export.py

```diff
@@ -106,10 +108,30 @@ async def export_virtualenv(
         if request.resolve
         else os.path.join("python", "virtualenv")
     )
+
+    merged_digest = await Get(Digest, MergeDigests([pex_pex.digest, requirements_pex.digest]))
+    pex_pex_path = os.path.join("{digest_root}", pex_pex.exe)
     return ExportResult(
         f"virtualenv for the resolve '{request.resolve}' (using {min_interpreter})",
         dest,
-        symlinks=[Symlink(venv_abspath, py_version)],
+        digest=merged_digest,
+        post_processing_cmds=[
+            PostProcessingCommand(
+                [
+                    pex_pex_path,
+                    os.path.join("{digest_root}", requirements_pex.name),
+                    "venv",
+                    "--pip",
+                    "--collisions-ok",
+                    "--remove=all",
+                    f"{{digest_root}}/{py_version}",
+                ],
+                {"PEX_MODULE": "pex.tools"},
+            ),
+            # Don't fail in the unlikely case that we don't have `rm` on the PATH.
+            # It's better to leave a stray pex.pex in dist/ than to error the entire export.
+            PostProcessingCommand(["rm", "-f", pex_pex_path, "||", "true"]),
```

Comment:
This can't work can it? `||` is a shellism and you're not running a shell. Isn't `-f` enough to always net success anyhow? It is for gnu `rm` anyhow.

Reply:
It is if `rm` exists, this is supposedly defense against `rm` not being on the path. I idly thought `||` was somehow a command, but you are of course right that it is a shellism.

Reply:
Let's wait until this turns out to be a problem before attempting to fix it. It seems pretty unlikely to be a worry in practice.

Reply:
If we have to we can add a flag to specify that a post-processing process is allowed to fail.

Reply:
PS this "worked" because `rm -f` saw `||` and `true` as arguments and silently failed to remove them.

Reply:
Yeah, I think hardcoding `rm` and expecting it to be in the PATH in an interactive process env is totally reasonable.

Reply:
> PS this "worked" ...

Nice.